sklearn/tree/_splitter.pyx

```diff
@@ -762,10 +775,10 @@ cdef class DensePartitioner:
         while p[0] + 1 < self.end and Xf[p[0] + 1] <= Xf[p[0]] + FEATURE_THRESHOLD:
             p[0] += 1
 
-        # (p + 1 >= end) or (Xf[p + 1] > X[p])
+        p_prev[0] = p[0]
+
+        # By adding 1, we have (Xf[p] >= end) or (Xf[p] > Xf[p - 1])
```

Comment:
Would it be better to change `Xf` everywhere in this file to `feature_values`? to be consistent?

Reply:
I thought it would be too long, but after trying it everywhere, it does not look too bad.